apps/train_loc/lib/train_loc/encoder/vehicle_positions_enhanced.ex

```diff
@@ -7,16 +7,47 @@ defmodule TrainLoc.Encoder.VehiclePositionsEnhanced do
 
   alias TrainLoc.Vehicles.Vehicle
 
+  @type feed() :: %{
+          header: feed_header(),
+          entity: [entity()]
+        }
+
+  @type feed_header() :: %{
+          gtfs_realtime_version: String.t(),
+          incrementality: integer(),
+          timestamp: integer()
+        }
+
+  @type entity() :: %{
+          id: non_neg_integer(),
+          vehicle: %{
+            trip:
+              %{start_date: String.t()} | %{start_date: String.t(), trip_short_name: String.t()},
+            vehicle:
+              %{id: non_neg_integer()} | %{id: non_neg_integer(), assignment_status: String.t()},
```

Comment:
suggestion: I believe you can do this with `required`/`optional`:
```elixir
%{
  required(:id) => non_neg_integer(),
  optional(:assignment_status) => String.t()
}
```

(`id` should also be a string, but it looks like it's not currently: that's not something you need to fix as a part of this PR.)